frontend/lib/color.js

```diff
@@ -0,0 +1,87 @@
+/**
```

Comment:
Do you think these methods will be reused outside of this field? I've been trying to keep component logic as self-contained as possible, as opposed to shared in lib files. Perhaps this could be moved to `frontend/components/FieldColor/util.js` and we can extract it later if we find the need to reuse this logic?